### PR TITLE
Add Gitlab deployconfig.

### PR DESCRIPTION
**Pull request by MichaelHatherly**
Just a quick exploratory PR to discuss whether we want to upstream this, since it makes running docs builds on gitlab CI and then uploading to gh pages easier. This is all the work of @jkrumbiegel, I'm just merely making the PR.

If we do what this then I'll add some tests and docs accordingly.

### CODE REVIEW

**Comment by mortenpi** [comment]
I think it would be great to have this.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Documenter.jl changelog
 
+## Version `v0.26.0`
+
+* ![Feature][badge-feature] Documenter can now deploy from Gitlab CI to GitHub Pages with `Documenter.Gitlab`.
+
```
Comment (MichaelHatherly) [inline CHANGELOG.md:6]: Marked as 0.26.0 since I assume we're not going to add this to a patch release?

Comment (fredrikekre) [reply CHANGELOG.md:6]: Pre 1.0 you should use the "patch number position" for both patches and features (https://julialang.github.io/Pkg.jl/v1/compatibility/#compat-pre-1.0-1).

Comment (MichaelHatherly) [reply CHANGELOG.md:6]: Ah, good to know, thanks.

**Comment by MichaelHatherly** [comment]
Linkcheck failure appears to be a gitlab URL. Gitlab is currently down as far as I can tell.

File: src/deployconfig.jl [hunk 2]
```diff
@@ -483,6 +483,160 @@ function post_github_status(type::S, deploydocs_repo::S, sha::S, subfolder=nothi
...
+
+ - `CI_COMMIT_TAG`: The commit tag name. Present only when building tags.
+
+ - `CI_PIPELINE_SOURCE`: Indicates how the pipeline was triggered.
+
+The `CI_*` variables are set automatically on Gitlab. More information on how Gitlab
+sets the `CI_*` variables can be found in the
+[Gitlab documentation](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html).
+"""
+struct Gitlab <: DeployConfig
```
Comment (mortenpi) [inline src/deployconfig.jl:518]: Very bikesheddy, but could we make it `GitLab`, with capital `L`?

File: src/deployconfig.jl [hunk 3]
```diff
@@ -483,6 +483,160 @@ function post_github_status(type::S, deploydocs_repo::S, sha::S, subfolder=nothi
...
+##########
+# Gitlab #
+##########
+
+"""
+    Gitlab <: DeployConfig
+
+Gitlab implementation of `DeployConfig`.
+
+The following environment variables influences the build when using the
```
Comment (mortenpi) [inline src/deployconfig.jl:495]:
```suggestion
The following environment variables influence the build when using the
```

File: src/deployconfig.jl [hunk 4]
```diff
@@ -483,6 +483,160 @@ function post_github_status(type::S, deploydocs_repo::S, sha::S, subfolder=nothi
...
+            io,
+            "- $(marker(tag_ok)) ENV[\"CI_COMMIT_TAG\"] contains a valid VersionNumber",
+        )
+        all_ok &= tag_ok
+
+        is_preview = false
+        subfolder = tag_nobuild
+        deploy_branch = branch
+        deploy_repo = repo
+
```
Comment (mortenpi) [inline src/deployconfig.jl:584]:
```suggestion
```

**Review by mortenpi — COMMENTED** [review]
Just a few cosmetic comments, otherwise LGTM!

**Comment by MichaelHatherly** [comment]
Comments addressed, should be good to go.

**Review by fredrikekre — APPROVED** [review]
Perhaps also modify https://github.com/JuliaDocs/Documenter.jl/pull/1448/files#diff-d03fe9850a3ee98315d4daf8671a29f0dc15348540762a9643d1cca14a5bc873R643 to autodetect GitLab?

**Comment by mortenpi** [comment]
LGTM, thanks!

**Pull request merged**